packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -0,0 +1,329 @@
+    // This array is parallel with the keys returned from `getRecordFieldNames()`.
+    readonly cellStateDataFieldNames: readonly string[];
+
+    // The keys from the row data that correlate to the data that will be in TCellData.
+    // This array is parallel with the keys specified by `cellStateDataFieldNames`.
+    abstract getRecordFieldNames(): string[];
+
+    // Function that allows the table column to validate the type that gets created
+    // for the cell data. This should validate that the types in TCellData are correct
+    // for each key defined by `cellStateDataFieldNames`.
+    // This function should throw if validation fails.
+    abstract validateCellData(cellData: TCellData): void;
+}
+```
+
+_Note: The `TableColumn` class may be updated to support other features not covered in this HLD such as sorting and grouping._
+
+Given the above class, a series of column elements to handle basic use cases can be written within Nimble. For example, the `TableColumn` implementation we could create for rendering data as a read-only `NimbleTextField` could look like this:
+
+```TS
+type TableColumnTextCellData = StringField<'value'>;
+type TableColumnTextColumnConfig = { placeholder: string };
+
+// this interface is used to pass auxiliary configuration to access within the cellTemplate
+interface TableColumnTextCellState<TCellData extends TableRecord, TableColumnTextColumnConfig> extends TableCellState<TCellData, TableColumnTextColumnConfig> { }
+
+public class TableColumnText extends TableColumn<TableColumnTextCellData, TableColumnTextColumnConfig> {
+    ...
+
+    public getColumnConfig(): TableColumnTextColumnConfig {
+        return { placeholder: this.placeholder };
+    }
+
+    public cellStateDataFieldNames = ['value'] as const;
+
+    @attr
+    public valueKey: string;
+
+    @attr
+    public placeholder: string; // Column auxiliary configuration
+
+    public getRecordFieldNames(): string[] {
+        return [valueKey];
+    }
+
+    public readonly cellTemplate: ViewTemplate<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>> =
+        html<TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>>`
+            <nimble-text-field readonly="true" value="${x => x.data.value}" placeholder="${x => x.columnConfig.placeholder}">
+            </nimble-text-field>
+        `;
+
+    public validateCellData(cellData: TCellData): void {
+        if(typeof(cellData['value']) !== 'string') {
+            throw new Error('Type for cellData is incorrect!');
+        }
+    }
+}
+```
+
+In the above example, notifications for when the `placeholder` property changed would be handled by the base class, and it would be responsible for any further action. This aciton could be a combination of things like causing a re-render, and issuing an event that may be publically visibile (such as `column-configuration-changed`). These details will be ironed out outside of this spec.
```

Comment:
```suggestion
In the above example, notifications for when the `placeholder` property changed would be handled by the base class, and it would be responsible for any further action. This action could be a combination of things like causing a re-render, and issuing an event that may be publicly visible (such as `column-configuration-changed`). These details will be ironed out outside of this spec.
```